pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -116,6 +116,9 @@ func (cgc *containerGC) enforceMaxContainersPerEvictUnit(evictUnits containersBy
 		toRemove := len(evictUnits[key]) - MaxContainers
 
 		if toRemove > 0 {
+			if len(evictUnits[key]) > toRemove {
```

Comment:
Could we instead move this to `removeOldestN` similar to `removeOldestNSandboxes`?